File: project_java/CommonLibrary/src/main/java/com/lzhs/order/commonlibrary/base/activity/BaseActivity.java
package com.lzhs.order.commonlibrary.base.activity;

import android.os.Bundle;

import com.lzhs.order.commonlibrary.base.mvp.BasePresenter;
import com.lzhs.order.commonlibrary.base.mvp.BaseView;
import com.lzhs.order.commonlibrary.di.commone.ActivityComponent;
import com.lzhs.order.commonlibrary.di.commone.DaggerActivityComponent;
import com.lzhs.order.commonlibrary.di.module.ActivityModule;

import javax.inject.Inject;

/**
 * Description: 描述 <br/>
 * Author: LZHS <br/>
 * Email: 1050629507@qq.com <br/>
 * Time: 2019/3/6 : 3:45 PM<br/>
 */
public abstract class BaseActivity<T extends BasePresenter> extends SimpleActivity implements BaseView {

    @Inject
    protected T mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initInject(DaggerActivityComponent.builder().activityModule(new ActivityModule(this))
                .appComponent(AppApplication.getAppComponent())
                .build());
        if (mPresenter != null) mPresenter.attachView(this);


    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mPresenter != null) {
            mPresenter.detachView();
            System.gc();
        }
    }

    public abstract void initInject(ActivityComponent activityComponent);
}

File: project_kotlin/CommonLibrary/src/main/java/com/lzhs/commonlibrary/bases/activity/BaseActivity.kt
package com.lzhs.commonlibrary.bases.activity

import com.lzhs.library.bases.BaseSwipeBackActivity

/**
 * Description: 描述 <br/>
 * Author: LZHS <br/>
 * Email: 1050629507@qq.com <br/>
 * Time: 2019/2/18 : 3:29 PM<br/>
 */
abstract class BaseActivity : BaseSwipeBackActivity() {
    override fun finish() {
        super.finish()
        overridePendingTransition(0, 0)
    }
}